lib/awful/client.lua.in

```diff
@@ -208,6 +208,7 @@ function client.focus.history.previous()
     end
     local c = client.focus.history.get(s, 1)
     if c then capi.client.focus = c end
```

Comment:
Please put the emit in the if and remove the old capi.client.focus, otherwise this will explode